src/components/textAreas/textAreaRichText/textAreaRichText.tsx

```diff
@@ -0,0 +1,115 @@
+import { Link } from '@tiptap/extension-link';
+import { Placeholder } from '@tiptap/extension-placeholder';
+import { EditorContent, useEditor } from '@tiptap/react';
+import { StarterKit } from '@tiptap/starter-kit';
+import classNames from 'classnames';
+import { useEffect, useId, useState } from 'react';
+import { createPortal } from 'react-dom';
+import { InputContainer, type IInputContainerProps } from '../../input';
+import { TextAreaRichTextActions } from './textAreaRichTextActions';
+
+export interface ITextAreaRichTextProps
+    extends Omit<IInputContainerProps, 'maxLength' | 'inputLength' | 'value' | 'onChange' | 'id'> {
+    /**
+     * Current value of the input.
+     */
+    value?: string;
+    /**
+     * Id of the input.
+     */
+    id?: string;
+    /**
+     * Callback called on value change.
+     */
+    onChange?: (value: string) => void;
+    /**
+     * Placeholder of the input.
+     */
+    placeholder?: string;
+}
+
+// Classes to properly style the TipTap placeholder
+// (see https://tiptap.dev/docs/editor/api/extensions/placeholder#additional-setup)
+const placeholderClasses = classNames(
+    'first:before:pointer-events-none first:before:absolute first:before:top-5', // General
+    'first:before:text-base/tight first:before:font-normal first:before:text-neutral-300', // Typography
+    'first:before:content-[attr(data-placeholder)]', // Content
+);
+
+export const TextAreaRichText: React.FC<ITextAreaRichTextProps> = (props) => {
+    const { value, onChange, placeholder, isDisabled, className, id, ...containerProps } = props;
+
+    const [isExpanded, setIsExpanded] = useState(false);
+
+    // Use random-id when id property is not specified for the input.
+    const randomId = useId();
+    const processedId = id ?? randomId;
+
+    const extensions = [
+        StarterKit,
+        Placeholder.configure({ placeholder, emptyNodeClass: placeholderClasses, showOnlyWhenEditable: false }),
+        Link,
+    ];
+
+    const editor = useEditor({
+        extensions,
+        content: value,
+        editable: !isDisabled,
+        editorProps: {
+            attributes: {
+                class: 'outline-none p-4 prose prose-neutral min-h-[144px] h-full max-w-none',
+                role: 'textbox',
```

Comment:
Is it possible to change the prose color to `neutral-600` or do you need to add a custom theme?

Reply:
No not really, what we can do is either choosing a grey scale between `slate`, `gray`, `neutral` or setup our own (https://tailwindcss.com/docs/typography-plugin#adding-custom-color-themes), will check later with Selim

Reply:
I was afraid of that lol